src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/DocumentBuilder.cs

```diff
@@ -0,0 +1,533 @@
+                    if (_docRoot.Usings.Count > 0)
+                    {
+                        var usingName = usingNode.Name.ToString();
+                        if (!_docRoot.Usings.Any(node => node.Name.ToString().Equals(usingName)))
+                        {
+                            _documentEditor.InsertAfter(_docRoot.Usings.Last(), usingNode);
+                        }
+                    }
+                }
+            }
+        }
+
+        //Add class members to the top of the class.
+        public ClassDeclarationSyntax AddProperties(ClassDeclarationSyntax classDeclarationSyntax)
+        { 
+            var modifiedClassDeclarationSyntax = classDeclarationSyntax;
+            if (_codeFile.ClassProperties != null && _codeFile.ClassProperties.Any() && classDeclarationSyntax != null)
+            {
+                //get a sample member for leading trivia. Trailing trivia will still be semi colon and new line.
+                var sampleMember = modifiedClassDeclarationSyntax.Members.FirstOrDefault();
+                var memberLeadingTrivia = sampleMember?.GetLeadingTrivia() ?? new SyntaxTriviaList(SyntaxFactory.Whitespace("    "));
+                var memberTrailingTrivia = new SyntaxTriviaList(SemiColonTrivia, SyntaxFactory.CarriageReturnLineFeed);
+
+                //create MemberDeclarationSyntax[] with all the Property strings. 
+                var classProperties = CreateClassProperties(modifiedClassDeclarationSyntax.Members, memberLeadingTrivia, memberTrailingTrivia);
+
+                if (classProperties.Length > 0)
+                {
+                    //add to the top of the class.
+                    var members = modifiedClassDeclarationSyntax.Members;
+                    foreach (var property in classProperties)
+                    {
+                        members = members.Insert(0, property);
+                    }
+                    modifiedClassDeclarationSyntax = modifiedClassDeclarationSyntax
+                        .WithMembers(members);
+                }
+            }
+            return modifiedClassDeclarationSyntax;
+        }
+
+        //Add class attributes '[Attribute]' to a class.
+        public ClassDeclarationSyntax AddClassAttributes(ClassDeclarationSyntax classDeclarationSyntax)
+        {
+            var modifiedClassDeclarationSyntax = classDeclarationSyntax;
+            if (_codeFile.ClassAttributes != null && _codeFile.ClassAttributes.Any() && classDeclarationSyntax != null)
+            {
+                var classAttributes = CreateAttributeList(_codeFile.ClassAttributes, modifiedClassDeclarationSyntax.AttributeLists, classDeclarationSyntax.GetLeadingTrivia());
+                modifiedClassDeclarationSyntax = modifiedClassDeclarationSyntax.WithAttributeLists(classAttributes);
+            }
+            return modifiedClassDeclarationSyntax;
+        }
+
+        public async Task WriteToClassFile(string fileName, string filePath)
```

Comment:
>WriteToClassFile [](http://example.com/codeflow?start=26&length=16)

WriteToClassFileAsync

Reply:
Also i see you never pass cancellation tokens in async methods , i assuem there is not cancellation scenario yet?

Reply:
fixed and moved https://github.com/dotnet/Scaffolding/pull/1593